packages/nx/src/plugins/js/index.ts

```diff
@@ -1,11 +1,78 @@
 import { ProjectGraphProcessor } from '../../config/project-graph';
 import { ProjectGraphBuilder } from '../../project-graph/project-graph-builder';
 import { buildNpmPackageNodes } from './project-graph/build-nodes/build-npm-package-nodes';
+import { buildExplicitDependencies } from './project-graph/build-dependencies/build-dependencies';
+import { readNxJson } from '../../config/configuration';
+import { fileExists, readJsonFile } from '../../utils/fileutils';
+import { PackageJson } from '../../utils/package-json';
+import {
+  lockFileExists,
+  lockFileHash,
+  parseLockFile,
+} from './lock-file/lock-file';
+import { NrwlJsPluginConfig, NxJsonConfiguration } from '../../config/nx-json';
+import { join } from 'path';
+import { projectGraphCacheDirectory } from '../../utils/cache-directory';
+import { readFileSync } from 'fs';
+import { workspaceRoot } from '../../utils/workspace-root';
 
-export const processProjectGraph: ProjectGraphProcessor = (graph) => {
+export const processProjectGraph: ProjectGraphProcessor = (graph, context) => {
   const builder = new ProjectGraphBuilder(graph);
 
+  const packageJson = readJsonFile<PackageJson>(
+    join(workspaceRoot, 'package.json')
+  );
+
+  const packageJsonDeps = {
+    ...packageJson.dependencies,
+    ...packageJson.devDependencies,
+  };
+
+  let lockHash = 'n/a';
+  // during the create-nx-workspace lock file might not exists yet
+  if (lockFileExists()) {
+    lockHash = lockFileHash();
+    if (getLastProcessedLockfileHash() !== lockHash) {
```

Comment:
why not just `getLastProcessedLockfileHash() !== lockFileHash()`. The `lockHash` is not used for anything else

Reply:
We need it to write the hash file :sweat_smile: